src/Algebra/Graph/Undirected.hs

```diff
@@ -0,0 +1,913 @@
+{-# LANGUAGE DeriveGeneric, RankNTypes #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Undirected
+-- Copyright  : (c) Andrey Mokhov 2016-2019
+-- License    : MIT (see the file LICENSE)
+-- Maintainer : andrey.mokhov@gmail.com
+-- Stability  : experimental
+--
+-- __Alga__ is a library for algebraic construction and manipulation of graphs
+-- in Haskell. See <https://github.com/snowleopard/alga-paper this paper> for the
+-- motivation behind the library, the underlying theory, and implementation details.
+--
+-- This module defines and abstraction over the core data type 'Graph' and associated algorithms.
+-- Undirected graphs satisfy all laws of the
+-- 'Algebra.Graph.Class.Undirected' type class, including the commutativity
+-- of 'connect'.
+-- 'Graph' is an instance of type classes defined in
+-- modules "Algebra.Graph.Class" and "Algebra.Graph.HigherKinded.Class", which
+-- can be used for polymorphic graph construction and manipulation.
+--
+-----------------------------------------------------------------------------
+module Algebra.Graph.Undirected (
+    -- * Algebraic data type for graphs
+    Graph, toGraph,
```

Comment:
I think we should rename `toGraph` to `fromUndirected` as in `Algebra.Graph.Relation.Symmetric`.

Reply:
Makes sense! I will change that :)